pallets/messages/src/lib.rs

```diff
@@ -165,6 +167,50 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Adds a message for an offchain resource
+		/// # Arguments
+		/// * `origin` - A signed transaction origin from the provider
+		/// * `on_behalf_of` - Optional. The msa id of delegate.
+		/// * `schema_id` - Registered schema id for current message.
+		/// * `cid` - The content address for an IPFS payload
+		/// * `payload_length` - The size of the payload
+		/// * Returns
+		/// * [DispatchResultWithPostInfo](https://paritytech.github.io/substrate/master/frame_support/dispatch/type.DispatchResultWithPostInfo.html)
+		#[pallet::weight(T::WeightInfo::add_ipfs_message(cid.get().len() as u32, 1_000))]
```

Comment:
is cid size dynamic? I don't think it is. If not it should not be used as in input here. We usually use it like this if the input is an array with dynamic size.

Reply:
I think the important thing in weight generation is that how much dynamic is the size of this cid? If its a difference between 32 bytes and 64 bytes then it's still close enough that I don't think will contribute significantly in weights. So the question might be what is the absolute min and max size for a CID.